contracts/factory/MagicDropCloneFactory.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {UUPSUpgradeable} from "solady/src/utils/UUPSUpgradeable.sol";
+import {Initializable} from "solady/src/utils/Initializable.sol";
+import {Ownable} from "solady/src/auth/Ownable.sol";
+import {LibClone} from "solady/src/utils/LibClone.sol";
+import {TokenStandard} from "../common/Structs.sol";
+import {MagicDropTokenImplRegistry} from "../registry/MagicDropTokenImplRegistry.sol";
+
+/// @title MagicDropCloneFactory
+/// @notice A factory contract for creating and managing clones of MagicDrop contracts
+/// @dev This contract uses the UUPS proxy pattern and is initializable
+contract MagicDropCloneFactory is Initializable, Ownable, UUPSUpgradeable {
```

Comment:
so do you recommend adding receive functions?

Reply:
nah that would just get more eth in the contracts. I think this can be fixed bumping open zeppelin contracts. But I haven't looked over all the changes. The current version is for sol 0.8.4. Latest OZ is 0.8.20 and doesn't have the payable functions. 

I guess there are a couple options -> Bumping OZ will effect the entire repo. Overriding all the functions with unintended behavior would not be fun. Accepting the risk of locked eth kind of leaves the problem as-is. And adding a withdrawal may be benign/never be used and is kind of just a patch. Interested in your thoughts, maybe I missed an option.

Reply:
I'm ok with the risk of locked funds